src/cloud-api-adaptor/test/provisioner/aws/provision_common.go

```diff
@@ -998,9 +999,12 @@ func ConvertQcow2ToRaw(qcow2 string, raw string) error {
 }
 
 // createCredentialFile Creates the AWS credential file in the install overlay directory
-// that's used by kustomize the setup CAA
-func createCredentialFile(dir, access_key_id, secret_access_key string) error {
+// that's used by kustomize the setup CAA. The session_token parameter is optional.
+func createCredentialFile(dir, access_key_id, secret_access_key, session_token string) error {
 	content := fmt.Sprintf("AWS_ACCESS_KEY_ID=%s\nAWS_SECRET_ACCESS_KEY=%s\n", access_key_id, secret_access_key)
+	if session_token != "" {
+		content += fmt.Sprintf("AWS_SESSION_TOKEN=%s\n", session_token)
+	}
```

Comment:
Does  session token need AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY ?

Reply:
if not, reflect the requirement for the token also in entrypoint.sh

Reply:
While session token is required for temporary credentials only, the access key and secret are *always* required.

Reply:
Ah so temporary credentials will not be an alternative to the access key and secret ?

Reply:
@wainersm as Snir mentioned, you'll need the AWS_SESSION_TOKEN param in the entrypoint.sh as well so that it can be set via peer-pods-cm

Reply:
> Ah so temporary credentials will not be an alternative to the access key and secret ?

It's not. You still need an access key and secret (which will eventually expire) and the session token is used to valid these credentials.

Reply:
> @wainersm as Snir mentioned, you'll need the AWS_SESSION_TOKEN param in the entrypoint.sh as well so that it can be set via peer-pods-cm

For what I've seen in entrypoint.sh the credentials (access key and secret) aren't passed as arguments to the caa program, but rather they are exported environment variables. In https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/entrypoint.sh#L54 it's checked that these variables are exported (or bail out) because access key and secrets are mandatory, but the session token is just optional.

But I might be missing something. @snir911 mind to explain?